cpp/include/cudf_test/stream_checking_resource_adaptor.hpp

```diff
@@ -33,7 +35,12 @@ class stream_checking_resource_adaptor final : public rmm::mr::device_memory_res
    *
    * @param upstream The resource used for allocating/deallocating device memory
    */
-  stream_checking_resource_adaptor(Upstream* upstream) : upstream_{upstream}
+  stream_checking_resource_adaptor(Upstream* upstream,
+                                   bool error_on_invalid_stream,
+                                   bool check_default_stream)
+    : upstream_{upstream},
```

Comment:
Why do we still use raw pointer? We should favor smart pointer, shouldn't we?

Reply:
If we cannot claim ownership of the pointer but have to use raw pointer, please add `CUDF_EXPECT` to check for nullptr.

Reply:
Smart pointers imply information about ownership semantics. When you want to be able to work with a pointer to some object regardless of who owns it, it's pretty normal to pass around raw pointers. One could argue that a reference would be better here to ensure that the parameter passed is non-null, but again I chose to follow the pattern used for other rmm adaptors. @jrhemstad or @harrism might have more insight into why a pointer was preferred to a reference here, I wouldn't expect adaptors to perform any of the types of modifications of the upstream where a pointer is preferred but I could be wrong.